lib/plaid/models.rb

```diff
@@ -736,5 +767,325 @@ class Transaction < BaseModel
+      property :available
+
+      ##
+      # :attr_reader:
+      # Public: The total amount of funds in the account.
+      property :current
+
+      ##
+      # :attr_reader:
+      # Public: The ISO currency code of the transaction, either USD or CAD.
+      # Always null if unofficial_currency_code is non-null.
+      property :iso_currency_code
+
+      ##
+      # :attr_reader:
+      # Public: The unofficial currency code associated with the transaction.
+      # Always null if iso_currency_code is non-null.
+      property :unofficial_currency_code
+    end
+
+    # Public: A representation of an asset report historical balance.
+    class AssetReportHistoricalBalance < BaseModel
+      ##
+      # :attr_reader:
+      # Public: The date of the calculated historical balance.
+      property :date
+
+      ##
+      # :attr_reader:
+      # Public: The total amount of funds in the account, calculated from the
+      # current balance in the balance object by subtracting inflows and adding
+      # back outflows according to the posted date of each.
+      property :current
+
+      ##
+      # :attr_reader:
+      # Public: The ISO currency code of the transaction, either USD or CAD.
+      # Always null if unofficial_currency_code is non-null.
+      property :iso_currency_code
+
+      ##
+      # :attr_reader:
+      # Public: The unofficial currency code associated with the transaction.
+      # Always null if iso_currency_code is non-null.
+      property :unofficial_currency_code
+    end
+
+    # Public: A representation of an asset report transaction.
+    class AssetReportTransaction < BaseModel
+      ##
+      # :attr_reader:
+      # Public: Plaid's unique identifier for the account.
+      property :account_id
+
+      ##
+      # :attr_reader:
+      # Public: Plaid's unique identifier for the transaction.
+      property :transaction_id
+
+      ##
+      # :attr_reader:
+      # Public: For pending transactions, Plaid returns the date the
+      # transaction occurred; for posted transactions, Plaid returns the date
+      # the transaction posts; both dates are returned in an ISO 8601 format
+      # (YYYY-MM-DD).
+      property :date
+
+      ##
+      # :attr_reader:
+      # Public: The string returned by the financial institution to describe
+      # the transaction.
+      property :original_description
+
+      ##
+      # :attr_reader:
+      # Public: When true, identifies the transaction as pending or unsettled;
+      # pending transaction details (description, amount) may change before
+      # they are settled.
+      property :pending
+
+      ##
+      # :attr_reader:
+      # Public: The settled dollar value; positive values when money moves out
+      # of the account, negative values when money moves in.
+      property :amount
+
+      ##
+      # :attr_reader:
+      # Public: The ISO currency code of the transaction, either USD or CAD.
+      # Always null if unofficial_currency_code is non-null.
```

Comment:
In Ruby speak this would sound as "Always nil if ...... non-nil".

Reply:
⚡️